src/__tests__/string.test.ts

```diff
@@ -262,3 +265,25 @@ test("datetime parsing", () => {
     datetimeOffset4Ms.parse("2020-10-14T17:42:29.124+00:00")
   ).toThrow();
 });
+
+test("numericString", () => {
+  const numericString = z.string().numericString();
+
+  expect(() => numericString.parse(true)).toThrow();
+  expect(() => numericString.parse(1)).toThrow();
+  expect(() => numericString.parse(undefined)).toThrow();
+  expect(() => numericString.parse(null)).toThrow();
+  expect(() => numericString.parse("")).toThrow();
+  expect(() => numericString.parse("hello")).toThrow();
+  expect(() => numericString.parse("3,1234")).toThrow();
+
+  expect(() => numericString.parse("1")).not.toThrow();
+  expect(() =>
+    numericString.parse(
+      "99999000000000000000000000000000000000000000000000000000"
+    )
+  ).not.toThrow();
+  expect(() => numericString.parse("3.141592")).not.toThrow();
+  expect(() => numericString.parse("-1")).not.toThrow();
+  expect(() => numericString.parse("-999.1231234134"));
+});
```

Comment:
A very minor (test) suggestion: as this `numericString` check appears to work for barcodes (i.e. a numeric string which is often zero padded, such as `0287730000000`), would it be worth adding a sample barcode to a test?

` expect(() => numericString.parse("0287730000000")).not.toThrow();`

Reply:
Sounds great. I added test case. Thanks for suggestion : )